cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -209,11 +234,17 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 		if err != nil {
 			return result, err
 		}
+
 		clientCACertFile := path.Join(s.SecureServing.ServerCert.CertDirectory, "client-ca.crt")
 		if err := os.WriteFile(clientCACertFile, testutil.EncodeCertPEM(clientSigningCert), 0644); err != nil {
 			return result, err
 		}
+
+		s.ProxyClientKeyFile = path.Join(s.SecureServing.ServerCert.CertDirectory, "misty-crt.key")
+		s.ProxyClientCertFile = path.Join(s.SecureServing.ServerCert.CertDirectory, "misty-crt.crt")
 		s.Authentication.ClientCert.ClientCA = clientCACertFile
+		s.Authentication.RequestHeader.ClientCAFile = proxyCACertFile
+		s.GenericServerRunOptions.PeerCAFile = path.Join(s.SecureServing.ServerCert.CertDirectory, s.SecureServing.ServerCert.PairName+".crt")
```

Comment:
I am reusing the self signed serving certs that also [contains the CA](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/util/cert/cert.go#L198-L204) for the test, since I couldnt find any other CA bundle that I could directly use here.